backend/src/db-utils.js

```diff
@@ -361,19 +518,206 @@ const queryActiveAccountsCountAggregatedByWeek = async () => {
   );
 };
 
+const queryIsAccountIndexed = async (accountId) => {
+  return await querySingleRow(
+    [
+      `SELECT account_id
+       FROM accounts
+       WHERE account_id = :accountId
+       LIMIT 1`,
+      { accountId },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryAccountsList = async (limit = 15, paginationIndexer) => {
+  return await queryRows(
+    [
+      `SELECT account_id AS account_id,
+              id AS account_index,
+              DIV(receipts.included_in_block_timestamp, 1000*1000) AS created_at_block_timestamp
+       FROM accounts
+       LEFT JOIN receipts ON receipts.receipt_id = accounts.created_by_receipt_id
+       ${paginationIndexer ? `WHERE id < :account_index` : ""}
+       ORDER BY account_index DESC
+       LIMIT :limit`,
+      {
+        limit,
+        account_index: paginationIndexer?.accountIndex,
+      },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+// this query is deprecated and will be removed
+const queryAccountTransactionsCount = async (accountId) => {
+  return await querySingleRow(
+    [
+      `SELECT
+       out_transactions_count.out_transactions_count,
+       in_transactions_count.in_transactions_count
+     FROM (SELECT
+           COUNT(transactions.transaction_hash) AS out_transactions_count
+         FROM transactions
+         WHERE signer_account_id = :accountId) AS out_transactions_count,
+         (SELECT
+           COUNT(DISTINCT transactions.transaction_hash) AS in_transactions_count
+         FROM transactions
+         LEFT JOIN receipts
+           ON receipts.originated_from_transaction_hash = transactions.transaction_hash
+         WHERE receipts.receiver_account_id = :accountId
+         AND transactions.signer_account_id != :accountId) AS in_transactions_count`,
+      { accountId },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryAccountOutcomeTransactionsCount = async (accountId) => {
+  async function queryOutcomeTransactionsCountFromAnalytics(accountId) {
+    const query = await querySingleRow(
+      [
+        `SELECT
+         SUM(outgoing_transactions_count) AS out_transactions_count
+         FROM daily_outgoing_transactions_per_account_count
+         WHERE account_id = :account_id
+         LIMIT 1`,
+        { account_id: accountId },
+      ],
+      { dataSource: DS_ANALYTICS_BACKEND }
+    );
+    if (!query || !query.out_transactions_count) {
+      return 0;
+    }
+    return parseInt(query.out_transactions_count);
+  }
+  async function queryTransactionsCountFromIndexerForLastDay(accountId) {
+    const query = await querySingleRow(
+      [
+        `SELECT
+         COUNT(transactions.transaction_hash) AS out_transactions_count
+         FROM transactions
+         WHERE signer_account_id = :account_id
+         AND transactions.block_timestamp >= (CAST(EXTRACT(EPOCH FROM DATE_TRUNC('day', NOW() - INTERVAL '1 day')) AS bigint) * 1000 * 1000 * 1000)
+         LIMIT 1`,
```

Comment:
1. Same comment about `LIMIT 1` + `COUNT` / `SUM` as above.
2. We discussed it offline, that we need to know the last computed date from analytics database in order to always compute the value correctly (otherwise, on the day boundary or when analytics database don't get updated for over a day we will have incorrect values). To query the latest date and the `SUM` in a single query you can use `MAX(collected_for_day)` next to `SUM`